features/IFC102_Absence-of-deprecated-entities.feature

```diff
@@ -1,74 +1,337 @@
+
+      Then <Attribute> = empty
+
+    Examples:
+      | Entity                        | Attribute                     |
+      | IfcFillAreaStyleHatching      | PointOfReferenceHatchLine     |
+
+
+  Scenario Outline: Check for deprecated enumerated values - IFC4.3
+    # ****** Description of deprecated enumerations for each entity *****
+    # IfcBuildingElementProxy: IfcBuildingElementProxyTypeEnum
+    # IfcFireSuppressionTerminal: IfcFireSuppressionTerminalTypeEnum
+    # IfcCableCarrierFitting: IfcCableCarrierFittingTypeEnum
+    # IfcGeographicElement: IfcGeographicElementTypeEnum
+    # IfcSpace: IfcSpaceTypeEnum
+    # IfcWall: IfcWallTypeEnum
+
+    Given A model with Schema "IFC4.3"
+    Given an <Entity>
+
+    Then PredefinedType is not <Value>
+
+    Examples:
+      | Entity                          |        Value                   |
+      | IfcBuildingElementProxy         | "PROVISIONFORSPACE"            |
+      | IfcBuildingElementProxy         | "PROVISIONFORVOID"             |
+      | IfcFireSuppressionTerminal      | "SPRINKLERDEFLECTOR"           |
+      | IfcFireSuppressionTerminalType  | "SPRINKLERDEFLECTOR"           |
+      | IfcCableCarrierFitting          | "TEE" or "CROSS" or "REDUCER"  |
+      | IfcCableCarrierFittingType      | "TEE" or "CROSS" or "REDUCER"  |
+      | IfcGeographicElement            | "SOIL_BORING_POINT"            |
+      | IfcGeographicElementType        | "SOIL_BORING_POINT"            |
+      | IfcSpace                        | "INTERNAL" or "EXTERNAL"       |
+      | IfcSpaceType                    | "INTERNAL" or "EXTERNAL"       |
+      | IfcStructuralLoadGroup          | "LOAD_COMBINATION"             |
+      | IfcWall                         | "POLYGONAL" or "STANDARD"      |
+      | IfcWallType                     | "POLYGONAL" or "STANDARD"      |
+
+
+  Scenario Outline: Check for deprecated enumerated values - IFC4
+    # IfcStructuralLoadGroup: IfcLoadGroupTypeEnum
+    # IfcWall: IfcWallTypeEnum
+
+    Given A model with Schema "IFC4"
+    Given an <Entity>
+
+    Then PredefinedType is not <Value>
+
+    Examples:
+      | Entity                   |  Value                                       |
+      | IfcStructuralLoadGroup   | "LOAD_COMBINATION"                           | 
+      | IfcWall                  | "POLYGONAL" or "STANDARD" or "ELEMENTEDWALL" |
+      | IfcWallType              | "POLYGONAL" or "STANDARD" or "ELEMENTEDWALL" |
+
+
+  Scenario Outline: Check for deprecated enumerated values - IFC2X3
+    # IfcOwnerHistory: IfcChangeActionEnum
+
+    Given A model with Schema "IFC2X3"
+    Given an <Entity>
+
+    Then <Attribute> is not <Value>
+
+    Examples:
+      | Entity                   | Attribute            |         Value                                |
+      | IfcOwnerHistory          | ChangeAction         | "MODIFIEDADDED" or "MODIFIEDDELETED"         |
+
+
+  Scenario Outline: Check for deprecated explicitly instantiated entities - IFC4.3
```

Comment:
Isn't enough the write this for IFC4, and then it applies to IFC4.3 too? It's the same set of entities, isn't it?

Reply:
Yes, that would be enough, but not consistent with the rest of the feature file.